app/controllers/home_controller.rb

```diff
@@ -1,4 +1,17 @@
 class HomeController < ApplicationController
   def index
+    @random_card = Card.where("review_date <= ?", Date.today).order("RANDOM()").limit(1)
   end
+
+  def check_translation
+    @card = Card.find(params[:card_id])
+    if @card.check_translation(params[:q])
+      flash[:notice] = "Правильный перевод"
+      @card.set_review_date
```

Comment:
you can update it directly from `set_review` date method
and you can move all business logic to model i thought